### PR TITLE
CVSB-914

### PR DESCRIPTION
**Pull request by mgherceanu**
I have changed the button to cancel on first page, then i have implemented the breadcrumbs feature, and also the text is now Capitalised as needed.

### CODE REVIEW

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts [hunk 1]
```diff
@@ -14,17 +14,31 @@ import { VehicleModel } from "../../../../models/vehicle/vehicle.model";
 export class TestTypesListPage implements OnInit {
   vehicleData: VehicleModel;
   testTypeReferenceData: TestTypesReferenceDataModel[];
+  firstPage: boolean;
+  previousPage;
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-types-list/test-types-list.ts]: define the types

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts [hunk 2]
```diff
@@ -14,17 +14,31 @@ import { VehicleModel } from "../../../../models/vehicle/vehicle.model";
...
   }
 
   ngOnInit() {
     this.getTestTypeReferenceData();
+    let previousView = this.navCtrl.getPrevious();
+    this.firstPage = previousView.id != 'TestTypesListPage';
+    if (this.firstPage) {
+      this.viewCtrl.setBackButtonText('Test type');
+    } else {
+      let backBtnName = this.backBtn.charAt(0).toUpperCase() + this.backBtn.slice(1, this.backBtn.length);
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-types-list/test-types-list.ts]: move it to common functions as it is used across the whole app

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts [hunk 3]
```diff
@@ -41,7 +55,9 @@ export class TestTypesListPage implements OnInit {
     if (testType.nextTestTypesOrCategories) {
       this.navCtrl.push('TestTypesListPage', {
         vehicleData: this.vehicleData,
-        testTypeData: testType.nextTestTypesOrCategories
+        testTypeData: testType.nextTestTypesOrCategories,
+        previousPage: testType.name,
+        backBtn: this.previousPage || 'Test type'
```
Comment (danieliordache) [inline src/pages/testing/test-creation/test-types-list/test-types-list.ts]: create/add 'test type' to an enum

Comment (mgherceanu) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: done

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts (hunk 2 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: done

File: src/pages/testing/test-creation/test-types-list/test-types-list.ts (hunk 1 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: done

Comment (mgherceanu) [reply src/pages/testing/test-creation/test-types-list/test-types-list.ts]: done

**Pull request closed without merging**